pytorch_lightning/profiler/pytorch.py

```diff
@@ -522,7 +507,7 @@ def _delete_profilers(self) -> None:
     def teardown(self, stage: Optional[str] = None) -> None:
         self._delete_profilers()
 
-        for k in self._recording_map:
+        for k in list(self._recording_map.keys()):
```

Comment:
nit:

```suggestion
        for k in self._recording_map:
```

Reply:
Because `self.stop(k)`  calls `del self._recording_map[action_name]`, if iterate using `for k in self._recording_map:` , will raise "RuntimeError: dictionary changed size during iteration". This code is refer to https://stackoverflow.com/questions/5384914/how-to-delete-items-from-a-dictionary-while-iterating-over-it